p4-16/spec/P4-16-spec.mdk

```diff
@@ -6620,6 +6620,33 @@ table is an abstract finite map that is given a key and produces as a
 result either an action or a "miss" indication, as described in
 Section [#sec-mau-semantics].
 
+The expected meaning of these values is as follows:
+
+- an `exact` match kind on a key field means that the value of the
+  field in the table specifies exactly all the bits for the key field.
+  This is applicable for key fields of all types supported by the target
+  which have an equality operation.
+
+- a `ternary` match kind on a key field means that the field in the
+  table specifies a set of values for the key field using a priority
+  and a mask.  A priority is a numeric value which is used to break
+  ties when a particular key belongs to multiple sets.  When table
+  entries are specified in the P4 program the priorities are generated
+  by the compiler; when entries are specified by the control-plane,
+  the priority may need to be explicitly specified.  Entries with
+  higher priority are matched first.  This specification does not
+  mandate whether "higher" priorities are represented by higher or
+  lower numeric values; this choice is left to the target
+  implementation.
+
+- a `lpm` (longest prefix match) match kind on a key field is a
+  specific type of `ternary` match where the mask is required to have
+  a form in binary that is a contiguous set of 1 bits followed by a
+  contiguous set of 0 bits.  Masks with more 1 bits have automatically
+  higher priorities.  A mask with all bits 0 is legal.
+
+An example specifying entries for a table is given in Section [#sec-entries]
+
```

Comment:
I do not know if this is the best place in the spec to say this, but it seems worth saying at least once somewhere in the spec something like the following:

"The text above defines the condition when one key lookup field matches one field within an entry added to the table.  If a table has multiple key field defined, then a lookup key only matches an entry if the match condition for all fields is true simultaneously for a single table entry."

I am betting there is probably a clearer way to state that.

Reply:
There is a sentence in the current PR addressing this comment, so I am resolving this.